src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -270,6 +273,17 @@ public class QueryInsightsSettings {
         Setting.Property.Dynamic
     );
 
+    /**
+     * Settings for the list of indices that excluded from top queries.
+     */
+    public static final Setting<List<String>> TOP_N_QUERIES_EXCLUDED_INDICES = Setting.listSetting(
```

Comment:
Add an update consumer for this setting. See `addSettingsUpdateConsumer` examples in `QueryInsightsService`